lib/gollum/app.rb

```diff
@@ -100,15 +91,35 @@ class App < Sinatra::Base
       @css = settings.wiki_options[:css]
       @js  = settings.wiki_options[:js]
       @mathjax_config = settings.wiki_options[:mathjax_config]
+
+      @use_static_assets = settings.wiki_options.fetch(:static, settings.environment == :production || settings.environment == :staging)
+      @static_assets_path = settings.wiki_options.fetch(:static_assets_path, './public/assets')
+
+      Sprockets::Helpers.configure do |config|
+        config.environment = settings.sprockets
+        config.prefix      = "#{@base_url}/#{Precious::Assets::ASSET_URL}"
+        config.digest      = @use_static_assets
+        if @use_static_assets
+          config.public_path = @static_assets_path
+          config.manifest = Sprockets::Manifest.new(settings.sprockets, @static_assets_path)
+        end
+      end
     end
 
     get '/' do
       redirect clean_url(::File.join(@base_url, @page_dir, wiki_new.index_page))
     end
 
     get '/assets/*' do
-      env['PATH_INFO'].sub!('/assets', '')
-      settings.sprockets.call(env)
+      env['PATH_INFO'].sub!("/#{Precious::Assets::ASSET_URL}", '')
+      if @use_static_assets
+        env['PATH_INFO'].sub!(Sprockets::Helpers.prefix, '') if @base_url
```

Comment:
This is a bit of an ugly hack, but I couldn't get it to work any other way. The problem is that when using static assets and `--base-path`, asset paths in the compiled CSS have a double prefix: `/tests/assets/test/assets`, for example. This is because we need to set the prefix to `test/assets` at compile-time, and at run time. Disabling either leads to paths that are malformed in different ways.

As far as I can see, the assets route now works both when using static and dynamic assets, and both with and without a base path.